File: OCH_Tao/ABC/C/365.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
  int64_t N, M, cnt;
  cin >> N >> M;
  cnt = M / N;
  vector<int64_t> A(N);
  for (int i = 0; i < N; i++)
  {
    cin >> A[i];
  }
  if (reduce(A.begin(), A.end()) <= M)
  {
    cout << "infinite" << endl;
  }
  else
  {
    while (true)
    {
      int64_t tmp = 0;
      for (int i = 0; i < N; i++)
      {
        tmp += min(cnt, A[i]);
      }
      if (tmp > M)
      {
        cout << cnt - 1 << endl;
        break;
      }
      else
      {
        cnt += 1;
      }
    }
  }
}

File: OCH_Tao/ABC/C/365.py
N,M = map(int,input().split())
A = sorted(list(map(int,input().split())),reverse=True)
cnt = M//N
if sum(A)<=M:
  print("infinite")
else:
  tmp = M-cnt*N
  if len([i-min(i,cnt) for i in A if i-min(i,cnt)!=0])<tmp:
    print(cnt)
  else:
    print(cnt+min([i-min(i,cnt) for i in A if i-min(i,cnt)!=0]))